packages/website/docs/sources.md

```diff
@@ -236,25 +260,158 @@ const autocomplete = createAutocomplete({
 });
 ```
 
-This pattern can be used to store data in the [context](context) before returning the sources.
+:::note
+
+You can use the official [`autocomplete-plugin-query-suggestions`](createQuerySuggestionsPlugin) plugin to retrieve Query Suggestions from Algolia.
+
+:::
+
+### Customizing items with templates
+
+In addition to defining data sources for items, a source also lets you customize how to display items using [`templates`](#templates-specific-to-algoliaautocomplete-js).
+
+:::info
+
+Templates are only supported in [autocomplete-js](autocomplete-js).
+
+:::
+
+Templates can return a string:
+
+```js
+import { autocomplete } from '@algolia/autocomplete-js';
+
+autocomplete({
+  // ...
+  getSources({ query }) {
+    return [
+      {
+        // ...
+        templates: {
+          item({ item }) {
+            return `Result: ${item.name}`;
+          },
+        },
+      },
+    ];
+  },
+});
+```
+
+Or a [Preact](https://preactjs.com/) component:
+
+```js
+import { autocomplete } from '@algolia/autocomplete-js';
+import { h } from 'preact';
+
+autocomplete({
+  // ...
+  getSources({ query }) {
+    return [
+      {
+        // ...
+        templates: {
+          item({ item }) {
+            return h('div', null, item.name);
```

Comment:
```suggestion
            return <div>{item.name}</div>;
            // This is equivalent to:
            // return h('div', null, item.name);
```

Reply:
Then we need to add a JSX pragma and we can drop the import of `h`, right?

Reply:
Handled in e9ea30d.